plugin-locationlayer/src/main/java/com/mapbox/mapboxsdk/plugins/locationlayer/StaleStateRunnable.java

```diff
@@ -0,0 +1,56 @@
+package com.mapbox.mapboxsdk.plugins.locationlayer;
+
+import android.os.Handler;
+
+import java.util.ArrayList;
+import java.util.List;
+
+class StaleStateRunnable implements Runnable {
+
+  private static StaleStateRunnable instance;
+
+  private static final int DEFAULT_DELAY_TIME_TILL_STALE = 10000;
+
+  private List<OnLocationStaleListener> onLocationStaleListeners;
+  private Handler handler;
+  private long delayTime = DEFAULT_DELAY_TIME_TILL_STALE;
+
+  static StaleStateRunnable getInstance() {
+    if (instance == null) {
+      instance = new StaleStateRunnable();
+    }
+    return instance;
+  }
+
+  private StaleStateRunnable() {
+    onLocationStaleListeners = new ArrayList<>();
+    handler = new Handler();
```

Comment:
these can be directly initialised as part of the field declaration + they can be made final